ce/artifacts/activation.ts

```diff
@@ -2,17 +2,35 @@
 // Licensed under the MIT License.
 
 import { delimiter } from 'path';
+import { Session } from '../session';
 import { linq } from '../util/linq';
 import { Uri } from '../util/uri';
 
 export class Activation {
+  constructor(protected session: Session) {
+  }
+
+  /** a collection of #define declarations that would assumably be applied to all compiler calls. */
   defines = new Map<string, string>();
-  tools = new Map<string, Uri>();
+
+  /** a collection of tool definitions from artifacts (think shell 'aliases')  */
+  tools = new Map<string, string>();
```

Comment:
Does there need to be getters for the tools? Or is that satisfied with the Variables() getter concatenating the tools and variables?

Reply:
It's a map -- you can do `activation.tools.get('python')` ... The `Variables ` property aggregates stuff up for the CLI activation experience. When I add in support for writing out aliases, I'll add another property accessor.